src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -792,6 +793,13 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                 return Failed;
             }
 
+            // merge the global analyzer and command line diagnostic options
+            if (analyzerConfigSet is object)
+            {
+                var diagnosticOptions = analyzerConfigSet.MergeGlobalTreeOptions(compilation.Options.SpecificDiagnosticOptions);
```

Comment:
This merge also needs to happen from Compilation options created from IDE. Will that be in a follow up change? Tagging @jasonmalinowski given he is doing something similar for tree diagnostic options now moving as well.

Reply:
Interesting, hadn't considered that. I can add it as part of this pr. Looks like I need to fix up some nullable warnings in the IDE anyway.

Reply:
If the merge logic can be moved into `CommandLineParser.Parse` so that `CommandLineArguments.CompilationOptions` has the merged specific diagnostic options, then IDE side doesn't need to do any work and also avoids exposing any public APIs from compiler to allow IDE to do this merge:

https://github.com/dotnet/roslyn/blob/fa69a8c9567246576fadcb13dc7eaccb748868e5/src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs#L282-L288

Should go before:
https://github.com/dotnet/roslyn/blob/fa69a8c9567246576fadcb13dc7eaccb748868e5/src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs#L139-L161

Reply:
As we discussed offline, best option to handle this merge for all analyzer execution scenarios outside of command line compiler is to merge `compilation.Options.SpecificDiagnosticOptions` and`analysisOptions.Options?.AnalyzerConfigOptionProvider.GlobalOptions.TreeOptions` dictionaries here:
https://github.com/dotnet/roslyn/blob/f4a14672ea85096f9c101961fd5ac96fe5482ad8/src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs#L129-L131